test/bluebird.test.js

```diff
@@ -94,10 +100,189 @@ describe('bluebird', function () {
         return Promise.reject();
       }, {
         max: 3,
+        stepbackBase: 0,
         timeout: 1500
       })).to.eventually.be.rejectedWith(Promise.TimeoutError).then(function () {
         expect(count).to.equal(3);
       });
     });
   });
-});
\ No newline at end of file
+  
+  describe('match', function () {
+    it('should continue retry while error is equal to match string', function () {
+        return expect(retry(function () {
+        count++;
+
+        if (count < 4) {
+          return Promise.reject(soRejected);
+        }
+
+        return Promise.resolve(soResolved);
+      }, {
+        max: 15,
+        stepbackBase: 1,
+        match: soRejected
+      })).to.eventually.be.equal(soResolved).then(function () {
+        expect(count).to.equal(4);
```

Comment:
Since you're testing a call count it might be better to just introduce a sinon stub here. (Goes for all tests that test call counts)